### PR TITLE
Limit the search tags to use from prerelease and preview versions.

### PR DESCRIPTION
**Pull request by isoos**
Related fix for #6352.

### CODE REVIEW

File: app/lib/search/backend.dart [hunk 1]
```diff
@@ -87,26 +87,24 @@ class SearchBackend {
...
 
-    // Find tags from latest prerelease and/or preview (if there one)
-    // This allows searching for tags with `<tag>-in-prerelease`.
-    // Example: `is:null-safe-in-prerelease`, or `platform:android-in-prerelease`
-    Future<List<String>> loadTags(String version) async {
-      final tags = <String>[];
+    // Find tags from latest prerelease and/or preview (if there one).
+    Future<Iterable<String>> loadTags(String version) async {
+      final tags = <String>{};
       final prv =
```
Comment (sigurdm) [inline app/lib/search/backend.dart:93]: Just a nit - could we un-abbreviate `prv` and `pra`.

File: pkg/_pub_shared/lib/search/tags.dart [hunk 2]
```diff
@@ -73,12 +76,27 @@ abstract class PackageVersionTags {
 
   /// Package version is compatible with Dart 3.
   static const String isDart3Ready = 'is:dart3-ready';
+
+  /// Version tags that provide a positive, forward-looking property
+  /// of a prerelease or preview version.
```
Comment (sigurdm) [inline pkg/_pub_shared/lib/search/tags.dart:81]: Can we extend this dartdoc with an explanation of why we only have the positive tags (eg. by explaining the consequence of having negative tags here).

**Review by sigurdm — APPROVED** [review]
LGTM

**Comment by sigurdm** [comment]
Can we also add a regression test?

**Review by jonasfj — APPROVED** [review]
We could also choose not to use tags from prereleases in search.

That makes our logic simpler, and easier to understand.

**Comment by isoos** [comment]
> We could also choose not to use tags from prereleases in search.

But then at the mids of the dart3 migration we will get this request again (as we did during the null-safety migration). Let's keep it, I think it is useful, but we need to filter the tags.

**Pull request merged**